### PR TITLE
[libsmb2] Bump to 2023-08-11

### PR DESCRIPTION
**Pull request by xiaozhuai**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

1. Bump to 2023-08-11
2. Fix macos build
3. Add usage

### CODE REVIEW

**Comment by LilyWangLL** [comment]
Thanks for your PR. The usage test failed, please add file `vcpkg-cmake-wrapper.cmake` to add `list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR})`.

**Comment by xiaozhuai** [comment]
> Thanks for your PR. The usage test failed, please add file vcpkg-cmake-wrapper.cmake to add list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR}).

@LilyWangLL 
Done~

**Comment by dg0yt** [comment]
Can you please double-check which CMake (config) files are installed, and where? It doesn't look reasonable at the moment.

Pulling Find modules in front of user-provided Find modules may break user projects and doesn't work outside vcpkg. Maybe the wrapper shouldn't be added here.

**Comment by xiaozhuai** [comment]
@dg0yt 
Here is the file list for `libsmb2:x64-osx-release`

```
x64-osx-release/
x64-osx-release/include/
x64-osx-release/include/smb2/
x64-osx-release/include/smb2/libsmb2-dcerpc-lsa.h
x64-osx-release/include/smb2/libsmb2-dcerpc-srvsvc.h
x64-osx-release/include/smb2/libsmb2-dcerpc.h
x64-osx-release/include/smb2/libsmb2-raw.h
x64-osx-release/include/smb2/libsmb2.h
x64-osx-release/include/smb2/smb2-errors.h
x64-osx-release/include/smb2/smb2.h
x64-osx-release/lib/
x64-osx-release/lib/libsmb2.a
x64-osx-release/lib/pkgconfig/
x64-osx-release/lib/pkgconfig/libsmb2.pc
x64-osx-release/share/
x64-osx-release/share/SMB2/
x64-osx-release/share/SMB2/vcpkg-cmake-wrapper.cmake
x64-osx-release/share/libsmb2/
x64-osx-release/share/libsmb2/FindSMB2.cmake
x64-osx-release/share/libsmb2/copyright
x64-osx-release/share/libsmb2/libsmb2-config-version.cmake
x64-osx-release/share/libsmb2/usage
x64-osx-release/share/libsmb2/vcpkg.spdx.json
x64-osx-release/share/libsmb2/vcpkg_abi_info.txt
```

**Comment by dg0yt** [comment]
So upstream produces a pointless `libsmb2-config-version.cmake`, and the port does a pointless fixup on this file...

File: ports/libsmb2/portfile.cmake [hunk 1]
```diff
@@ -3,23 +3,26 @@ vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
...
+vcpkg_cmake_config_fixup(CONFIG_PATH "lib/cmake/${PORT}")
 vcpkg_fixup_pkgconfig()
 
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
-
-#the debug/share folder is generated empty by the provided cmake system
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
 
-file(INSTALL "${SOURCE_PATH}/COPYING" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
+file(INSTALL "${CMAKE_CURRENT_LIST_DIR}/vcpkg-cmake-wrapper.cmake" DESTINATION "${CURRENT_PACKAGES_DIR}/share/SMB2")
```
Comment (dg0yt) [inline ports/libsmb2/portfile.cmake:25]: We prefer lower-case dirs in `share`.
(Some people even want to add a CI check for it.)

File: ports/libsmb2/usage [hunk 2]
```diff
@@ -0,0 +1,5 @@
+The package libsmb2 provides CMake targets:
```
Comment (dg0yt) [inline ports/libsmb2/usage:1]: I don't see any targets here.

File: ports/libsmb2/portfile.cmake (hunk 1 above)
Comment (xiaozhuai) [reply ports/libsmb2/portfile.cmake:25]: > Some people even want to add a CI check for it

That would be better : )

File: ports/libsmb2/usage (hunk 2 above)
Comment (xiaozhuai) [reply ports/libsmb2/usage:1]: Ah... I copy it from other usage file, will change it.

**Comment by xiaozhuai** [comment]
@dg0yt 
Done~
```
x64-osx-release/
x64-osx-release/include/
x64-osx-release/include/smb2/
x64-osx-release/include/smb2/libsmb2-dcerpc-lsa.h
x64-osx-release/include/smb2/libsmb2-dcerpc-srvsvc.h
x64-osx-release/include/smb2/libsmb2-dcerpc.h
x64-osx-release/include/smb2/libsmb2-raw.h
x64-osx-release/include/smb2/libsmb2.h
x64-osx-release/include/smb2/smb2-errors.h
x64-osx-release/include/smb2/smb2.h
x64-osx-release/lib/
x64-osx-release/lib/libsmb2.a
x64-osx-release/lib/pkgconfig/
x64-osx-release/lib/pkgconfig/libsmb2.pc
x64-osx-release/share/
x64-osx-release/share/libsmb2/
x64-osx-release/share/libsmb2/copyright
x64-osx-release/share/libsmb2/usage
x64-osx-release/share/libsmb2/vcpkg.spdx.json
x64-osx-release/share/libsmb2/vcpkg_abi_info.txt
x64-osx-release/share/smb2/
x64-osx-release/share/smb2/FindSMB2.cmake
x64-osx-release/share/smb2/libsmb2-config-version.cmake
x64-osx-release/share/smb2/vcpkg-cmake-wrapper.cmake
```

**Review by LilyWangLL — APPROVED** [review]

**Comment by LilyWangLL** [comment]
Usage test passed on `x64-windows`.

**Comment by BillyONeal** [comment]
Thanks for the update!

**Pull request merged**